google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineAction.java

```diff
@@ -65,14 +68,29 @@ public AppEngineAction(
     this.callback = callback;
   }
 
+  /**
+   * Returns the commandline process handler used to execute the action
+   *
+   * @return the process handler, or null, if the action is not executing
+   */
+  public ProcessHandler getProcessHandler() {
+    return processHandler;
+  }
+
   protected LoggingHandler getLoggingHandler() {
     return loggingHandler;
   }
 
-  protected void executeProcess(
+  protected synchronized void executeProcess(
```

Comment:
I think you're right, this is not needed because we have synchronized on AppEngineRuntimeInstance.deploy() and .disconnect() where the lifecycle of the action is managed.